File: web/src/core/rules/bucketKey.ts
// Spec: docs/specs/rules/bucket-key.md
// Mudanças aqui DEVEM começar pela spec.

import type { Row } from "../types";

export function bucketKey(row: Row): string {
  if (row.origem === "Cartão") {
    return row.rateio === "Metade"
      ? "Cartão (compartilhado)"
      : "Cartão (pessoal)";
  }
  return row.origem;
}

File: app/lib/core/rules/bucket_key.dart
// Spec: docs/specs/rules/bucket-key.md
// Mudanças aqui DEVEM começar pela spec.

import '../types.dart';

String bucketKey(ExpenseRow row) {
  if (row.origem == 'Cartão') {
    return row.rateio == 'Metade'
        ? 'Cartão (compartilhado)'
        : 'Cartão (pessoal)';
  }
  return row.origem;
}
